src/lucky/page_helpers/svg_inliner.cr

```diff
@@ -0,0 +1,27 @@
+annotation Lucky::SvgInliner::Path
+end
+annotation Lucky::SvgInliner::StripRegex
+end
+
+@[Lucky::SvgInliner::Path("src/svgs")]
+@[Lucky::SvgInliner::StripRegex(/(class|fill|stroke|stroke-width|style)="[^"]+" ?/)]
```

Comment:
Can you help me understand what's going on with these annotations? It seems like a roundabout way to accomplish arguments to the macro.

Reply:
Sure. I think having those parameters passed as arguments would be tedious. Let's say the user wants to use a different directory for the SVG files, then it suffices to override the annotation in an initializer:

```crystal
# config/svg_inliner.cr
@[Lucky::SvgInliner::Path("src/icons")]
module Lucky::SvgInliner
end
```

As opposed to (for every single svg):

```crystal
inline_svg("menu-social/twitter.svg", false, "src/icons")
inline_svg("menu-social/discord.svg", false, "src/icons")
```

I prefer to use annotations instead of constants because they're more flexible. You wouldn't be able to override a constant in the same way.

Does that answer your question? Or am I missing the point?

Reply:
Oh interesting, I hadn't considered that. I'm not opposed. Another possibility for this would be to have a Habitat config parameter. @jwoertink @stephendolan thoughts?

Reply:
Tried that first, but settings from Habitat are not available at compile-time. As far as I know, the only way to allow compile-time configuration is by using annotations of constants.  And using constants is not favourable because it's impossible to override them.

I think these are sensible defaults, so in 99% of the cases, it wouldn't be necessary to change them. The annotations are just an escape hatch.

Reply:
@wout yeah, you present a use case I can't argue with. Thank you for walking me through it. Let's go!